src/metax_api/tests/api/rest/base/views/datasets/write.py

```diff
@@ -3899,6 +3914,97 @@ def test_user_projects_are_checked_when_writing_into_protected_data_catalog(self
+        del self.cr_test_data['research_dataset']['total_files_byte_size']
+
+        self._use_http_authorization(username=django_settings.API_EXT_USER['username'],
+            password=django_settings.API_EXT_USER['password'])
+
+    def test_external_service_can_add_cr_to_own_catalog(self):
+        self.cr_test_data['data_catalog'] = EXT_CATALOG
+        self.cr_test_data['research_dataset']['preferred_identifier'] = '123456'
+        response = self.client.post('/rest/datasets', self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+        self.assertEqual(response.data['research_dataset']['preferred_identifier'], '123456')
+
+    def test_external_service_can_update_cr_in_own_catalog(self):
+        self.cr_test_data['data_catalog'] = EXT_CATALOG
+        self.cr_test_data['research_dataset']['preferred_identifier'] = '123456'
+        response = self.client.post('/rest/datasets', self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+        self.assertEqual(response.data['research_dataset']['preferred_identifier'], '123456')
+
+        cr_id = response.data['id']
+        self.cr_test_data['research_dataset']['preferred_identifier'] = '654321'
+        response = self.client.put('/rest/datasets/{}'.format(cr_id), self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(response.data['research_dataset']['preferred_identifier'], '654321')
+
+    def test_external_service_can_not_add_cr_to_other_catalog(self):
+        dc = self._get_object_from_test_data('datacatalog', requested_index=1)
+        self.cr_test_data['data_catalog'] = dc['catalog_json']['identifier']
+        response = self.client.post('/rest/datasets', self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
+    def test_external_service_can_not_update_cr_in_other_catalog(self):
+        self._use_http_authorization()
+        dc = self._get_object_from_test_data('datacatalog', requested_index=1)
+        self.cr_test_data['data_catalog'] = dc['catalog_json']['identifier']
+        response = self.client.post('/rest/datasets', self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+
+        cr_id = response.data['id']
+        self._use_http_authorization(username=django_settings.API_EXT_USER['username'],
+            password=django_settings.API_EXT_USER['password'])
+        self.cr_test_data['research_dataset']['preferred_identifier'] = '654321'
+        response = self.client.put('/rest/datasets/{}'.format(cr_id), self.cr_test_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
+    def test_external_service_not_creating_cr_identifier(self):
```

Comment:
The name ``_cr_identifier`` is probably a typo, ``preferred_identifier`` is meant?

Reply:
i meant catalog record identifier, should probably correct for clarity

Reply:
``cr_identifier`` typically means the ``cr.identifier`` field, which is Metax internal, and is never created by anyone else than Metax. ``cr.research_dataset.preferred_identifier`` on the otherhand is provided by the harvesting party, but can also be generated by Metax, when the dataset is brand new and does not have any existing external idenitifer yet.

The test anyway seems to be testing ``preferred_identifier``, and not the internal identifier-field, so the test title is misleading here.